network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/extensions/CgmesControlAreaImpl.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.network.store.iidm.impl.extensions;
+
+import com.powsybl.cgmes.extensions.CgmesControlArea;
+import com.powsybl.iidm.network.*;
+import com.powsybl.network.store.iidm.impl.NetworkImpl;
+import com.powsybl.network.store.iidm.impl.TerminalRefUtils;
+import com.powsybl.network.store.model.CgmesControlAreaAttributes;
+
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+public class CgmesControlAreaImpl implements CgmesControlArea {
+
+    private final NetworkImpl network;
+
+    private final CgmesControlAreaAttributes attributes;
+
+    CgmesControlAreaImpl(NetworkImpl network, CgmesControlAreaAttributes attributes) {
+        this.network = Objects.requireNonNull(network);
+        this.attributes = Objects.requireNonNull(attributes);
+    }
+
+    @Override
+    public String getId() {
+        return attributes.getId();
+    }
+
+    @Override
+    public String getName() {
+        return attributes.getName();
+    }
+
+    @Override
+    public String getEnergyIdentificationCodeEIC() {
+        return attributes.getEnergyIdentificationCodeEic();
+    }
+
+    @Override
+    public Set<Terminal> getTerminals() {
+        return attributes.getTerminals().stream().map(a -> TerminalRefUtils.getTerminal(network.getIndex(), a)).collect(Collectors.toSet());
+    }
+
+    public static Boundary getTerminalBoundary(Terminal terminal) {
+        if (terminal.getConnectable() instanceof DanglingLine) {
+            return ((DanglingLine) terminal.getConnectable()).getBoundary();
+        } else if (terminal.getConnectable() instanceof TieLine) {
+            TieLine tieLine = (TieLine) terminal.getConnectable();
+            Branch.Side side = terminal == tieLine.getTerminal1() ? Branch.Side.ONE : Branch.Side.TWO;
+            return ((TieLine) terminal.getConnectable()).getHalf(side).getBoundary();
```

Comment:
Here, you can use variable 'tieLine' initialized above

Reply:
Done